pkg/network/setup/netstat_test.go

```diff
@@ -329,6 +329,90 @@ var _ = Describe("netstat", func() {
+				newVMISpecPodNetwork(primaryNetworkName),
+				newDomainSpecIface(primaryNetworkName, primaryMAC),
+				primaryPodIPv4, primaryPodIPv6,
+			)
+			setup.addNetworkInterface(
+				newVMISpecIfaceWithBridgeBinding(secondaryNetworkName),
+				newVMISpecMultusNetwork(secondaryNetworkName),
+				newDomainSpecIface(secondaryNetworkName, secondaryMAC),
+				secondaryPodIPv4, secondaryPodIPv6,
+			)
+		})
+
+		It("changes the MAC addresses of the masquerade (pod network) and bridge interfaces", func() {
+			setup.addGuestAgentInterfaces(
+				newDomainStatusIface(primaryIfaceName, nil, primaryMAC, "", netvmispec.InfoSourceDomain),
+				newDomainStatusIface(secondaryIfaceName, nil, secondaryMAC, "", netvmispec.InfoSourceDomain),
+				newDomainStatusIface("", []string{primaryGaIPv4, primaryGaIPv6}, newMAC1, primaryIfaceName, netvmispec.InfoSourceGuestAgent),
+				newDomainStatusIface("", []string{secondaryGaIPv4, secondaryGaIPv6}, newMAC2, secondaryIfaceName, netvmispec.InfoSourceGuestAgent),
+			)
+			setup.NetStat.UpdateStatus(setup.Vmi, setup.Domain)
+
+			Expect(setup.Vmi.Status.Interfaces).To(ConsistOf([]v1.VirtualMachineInstanceNetworkInterface{
+				newVMIStatusIface(primaryNetworkName, []string{primaryPodIPv4, primaryPodIPv6}, primaryMAC, "", netvmispec.InfoSourceDomain),
+				newVMIStatusIface(secondaryNetworkName, nil, secondaryMAC, "", netvmispec.InfoSourceDomain),
+				newVMIStatusIface("", []string{primaryGaIPv4, primaryGaIPv6}, newMAC1, primaryIfaceName, netvmispec.InfoSourceGuestAgent),
+				newVMIStatusIface("", []string{secondaryGaIPv4, secondaryGaIPv6}, newMAC2, secondaryIfaceName, netvmispec.InfoSourceGuestAgent),
+			}))
+		})
+
+		It("creates a new interface", func() {
+			const (
+				newGaIPv4    = "3.3.3.3"
+				newGaIPv6    = "fd20:333::3333"
+				newIfaceName = "eth3"
+			)
+			setup.addGuestAgentInterfaces(
+				newDomainStatusIface(primaryNetworkName, []string{primaryGaIPv4, primaryGaIPv6}, primaryMAC, primaryIfaceName, netvmispec.InfoSourceDomainAndGA),
+				newDomainStatusIface(secondaryNetworkName, []string{secondaryGaIPv4, secondaryGaIPv6}, secondaryMAC, secondaryIfaceName, netvmispec.InfoSourceDomainAndGA),
+				newDomainStatusIface("", []string{newGaIPv4, newGaIPv6}, newMAC1, newIfaceName, netvmispec.InfoSourceGuestAgent),
+			)
+			setup.NetStat.UpdateStatus(setup.Vmi, setup.Domain)
+
+			Expect(setup.Vmi.Status.Interfaces).To(ConsistOf([]v1.VirtualMachineInstanceNetworkInterface{
+				newVMIStatusIface(primaryNetworkName, []string{primaryPodIPv4, primaryPodIPv6}, primaryMAC, primaryIfaceName, netvmispec.InfoSourceDomainAndGA),
+				newVMIStatusIface(secondaryNetworkName, []string{secondaryPodIPv4, secondaryPodIPv6}, secondaryMAC, secondaryIfaceName, netvmispec.InfoSourceDomainAndGA),
+				newVMIStatusIface("", []string{newGaIPv4, newGaIPv6}, newMAC1, newIfaceName, netvmispec.InfoSourceGuestAgent),
+			}))
+		})
+
+		It("make the existing interfaces disappear", func() {
```

Comment:
disappear?
ain't shows exactly the domain interfaces that we have on the BeforeEach ?

Reply:
> maybe worth to add in the same test a scenario where they appear, and then to remove them and see they gone ?
unless not needed to check the function one instance after another in the same test

The test is checking a scenario based on a given input and a result of processing it.
There is no state mutated between updates, so I see no need to do this.

> ok i understand it shows the case where the GA doesn't report them, maybe worth to write that down as part of the desc ?

I do not think this is more complex than the others. Here we do not place a GA input, which simulates that no interfaces exist in the guest.

Reply:
> I do not think this is more complex than the others. Here we do not place a GA input, which simulates that no interfaces exist in the guest.

I find it not enough explanatory, as "disappear" says they were and gone, but here they were not even,
so i think it would worth to explain it a bit better than the current desc
Something like "When no guest agent interfaces are reported, should include only the domain interfaces"

Reply:
So you are suggesting to change the test name/text, not to add comments to explain it.
I changed it, hopefully it is more clear now.